press/api/callbacks.py

```diff
@@ -41,42 +42,50 @@ def verify_job_id(server: str, job_id: str):
 
 
 def handle_job_updates(server: str, job_identifier: str):
-	job_id = job_identifier
-	server: Server = frappe.get_doc("Server", server)
-	agent = server.agent
-	press_settings: PressSettings = frappe.get_doc("Press Settings")
-
-	if not press_settings.use_agent_job_callbacks or not server.use_agent_job_callbacks:
-		return
-
-	# For some reason output is not returned when job returns from rq callback
-	polled_job = agent.get_job_status(job_id)
-
-	job = frappe.get_value(
-		"Agent Job",
-		fieldname=[
-			"name",
-			"job_id",
-			"status",
-			"callback_failure_count",
-			"job_type",
-		],
-		filters={"job_id": job_id},
-		as_dict=True,
-	)
-
-	callback = frappe.get_doc(
-		{
-			"doctype": "Agent Job Callback",
-			"job_name": job.job_type,
-			"agent_job": polled_job["agent_job_id"],
-			"status": polled_job["status"],
-		}
-	)
-	callback.insert()
-	frappe.db.commit()
-
-	handle_polled_job(polled_job=polled_job, job=job)
+	current_user = frappe.session.user
+	try:
+		frappe.set_user("Administrator")
+		job_id = job_identifier
+		server: Server = frappe.get_doc("Server", server)
+		agent = server.agent
+		press_settings: PressSettings = frappe.get_doc("Press Settings")
+
+		if not press_settings.use_agent_job_callbacks or not server.use_agent_job_callbacks:
+			return
+
+		# For some reason output is not returned when job returns from rq callback
+		polled_job = agent.get_job_status(job_id)
+
+		job = frappe.get_value(
+			"Agent Job",
+			fieldname=[
+				"name",
+				"job_id",
+				"status",
+				"callback_failure_count",
+				"job_type",
+			],
+			filters={"job_id": job_id},
+			as_dict=True,
+		)
+
+		callback = frappe.get_doc(
+			{
+				"doctype": "Agent Job Callback",
+				"job_name": job.job_type,
+				"agent_job": polled_job["agent_job_id"],
+				"status": polled_job["status"],
+			}
+		)
+		callback.insert()
+		frappe.db.commit()
```

Comment:
This `handle_job_updates` func is running in background job.
Avoid manual commit then, because `handle_polled_job` func will run next.